sparkly/session.py

```diff
@@ -187,11 +188,20 @@ def _setup_jars(self):
             return ''
 
     def _setup_options(self, additional_options):
-        options = list(self.options.items())
+        options = {}
+
+        options.update(self.options)
+
         if additional_options:
-            options += list(additional_options.items())
+            options.update(additional_options)
 
-        return sorted(options)
+        if 'spark.sql.catalogImplementation' not in options:
+            options['spark.sql.catalogImplementation'] = 'hive'
+
+        if options:
```

Comment:
nit: the if is `unnecessary` since it will give the same result in the empty case.

Reply:
Was copy-pasting from the functions above.